File: running-sum-array/RunningSumArray.swift
import Foundation

class RunningSumArray {
    func runningSum(_ nums: [Int]) -> [Int] {
        var result: [Int] = Array(repeating: 0, count: nums.count)
        result[0] = nums[0]
        for index in 1 ..< nums.count {
            result[index] = result[index - 1] + nums[index]
        }
        return result
    }
}

File: running-sum-array/RunningSumArray.cs
public class RunningSumArray {
    public int[] RunningSum(int[] nums) {
        int[] result = new int[nums.Length];
        result[0] = nums[0];
        int index = 1;
        while(index < nums.Length) {
            result[index] = result[index - 1] + nums[index];
            index++;
        }
        return result;
    }
}